lib/Fmt_ast.ml

```diff
@@ -2794,78 +2782,79 @@ and fmt_cases c ctx cs =
-      let indent =
-        match
-          (c.conf.cases_matching_exp_indent, (ctx, pc_rhs.pexp_desc))
-        with
-        | ( `Compact
-          , ( Exp {pexp_desc= Pexp_function _ | Pexp_match _ | Pexp_try _; _}
-            , (Pexp_match _ | Pexp_try _) ) ) ->
-            2
-        | _, _ -> c.conf.cases_exp_indent
-      in
-      let align_nested_match =
-        match (pc_rhs.pexp_desc, c.conf.nested_match) with
-        | (Pexp_match _ | Pexp_try _), `Align -> last
-        | _ -> false
-      in
-      let parens_here, parens_for_exp =
-        if align_nested_match then (false, Some false)
-        else if c.conf.leading_nested_match_parens then (false, None)
-        else (parenze_exp xrhs, Some false)
-      in
-      (* side effects of Cmts.fmt_before before [fmt_lhs] is important *)
-      let leading_cmt = Cmts.fmt_before c pc_lhs.ppat_loc in
-      let xlhs = sub_pat ~ctx pc_lhs in
-      let paren_lhs =
-        match pc_lhs.ppat_desc with
-        | Ppat_or _ when Option.is_some pc_guard -> true
-        | _ -> parenze_pat xlhs
-      in
-      let eol =
-        Option.some_if
-          (Cmts.has_before c.cmts pc_rhs.pexp_loc)
-          (fmt "@;<1000 0>")
-      in
-      let indent = if align_nested_match then 0 else indent in
-      let fmt_padding =
-        let level = match c.conf.break_cases with `Nested -> 2 | _ -> 3 in
-        fmt_if_k
-          ( c.conf.align_cases
-          && not (Cmts.has_after c.cmts xlhs.ast.ppat_loc) )
-          ( match (max_len_name, pattern_len case) with
-          | Some max_len, Some len ->
-              let pad = String.make (max_len - len) ' ' in
-              fmt_or_k
-                Poly.(c.conf.break_cases = `All)
-                (str pad)
-                (fits_breaks ~level "" pad)
-          | _ -> noop )
-      in
-      Params.get_cases c.conf ~first ~indent ~parens_here
-      |> fun (p : Params.cases) ->
-      p.leading_space $ leading_cmt
-      $ p.box_all
-          ( p.box_pattern_arrow
-              ( hvbox 0
-                  ( fmt_pattern c ~pro:p.bar ~parens:paren_lhs xlhs
-                  $ fmt_padding
-                  $ opt pc_guard (fun g ->
-                        fmt "@;<1 2>when "
-                        $ fmt_expression c (sub_exp ~ctx g)) )
-              $ p.break_before_arrow $ str "->" $ p.break_after_arrow
-              $ fmt_if parens_here " (" )
-          $ p.break_after_opening_paren
-          $ hovbox 0
-              ( fmt_expression ?eol c ?parens:parens_for_exp xrhs
-              $ fmt_if parens_here
-                  ( match c.conf.indicate_multiline_delimiters with
-                  | `Space -> "@ )"
-                  | `No -> "@,)"
-                  | `Closing_on_separate_line -> "@;<1000 -2>)" ) ) ))
+  let max_len = fold_pattern_len ~f:max cs in
+  list_fl cs (fun ~first ~last case ->
+      let pattern_len = pattern_len case in
+      fmt_case c ctx ~first ~last ~max_len ~pattern_len case)
```

Comment:
You could pass the padding here (`max_len - pattern_len`) instead. So there is less argument and we avoid a match expecting 2 `option` to be in sync.

Reply:
agreed, fixed but could not simplify as much as you wanted.
To make it easier to read `break-infix-before-func=false` could be added in our `.ocamlformat`

Reply:
Tweaking the `.ocamlformat` is not necessary. My comment was only about separating concerns.
To simplify the code, I think matching on `max_len` and `pattern_len` is easier to read than bind operators. You could also use a `when` clause to disable padding.

Reply:
done